src/objects/zcl_abapgit_object_intf.clas.locals_imp.abap

```diff
@@ -609,16 +618,77 @@ CLASS lcl_aff_metadata_handler IMPLEMENTATION.
     CREATE OBJECT lo_ajson.
     TRY.
         lo_ajson->deserialize(
-           EXPORTING
-             iv_content = iv_data
-             iv_defaults = lt_values_for_initial
-             iv_enum_mappings = lt_enum_mappings
-           IMPORTING
-             ev_data    = rv_result ).
+          EXPORTING
+            iv_content       = iv_data
+            iv_defaults      = lt_values_for_initial
+            iv_enum_mappings = lt_enum_mappings
+          IMPORTING
+            ev_data          = rv_result ).
       CATCH cx_static_check INTO lx_exception.
         zcx_abapgit_exception=>raise_with_text( lx_exception ).
     ENDTRY.
 
   ENDMETHOD.
 
+  METHOD serialize_translation.
+    DATA: ls_data        TYPE zif_abapgit_aff_intf_v1=>ty_main,
+          lv_langu       TYPE laiso,
+          lv_json        TYPE string,
+          lo_ajson       TYPE REF TO zif_abapgit_ajson,
+          lo_json_path   TYPE REF TO zcl_abapgit_json_path,
+          lt_translation TYPE string_table,
+          lx_exception   TYPE REF TO zcx_abapgit_ajson_error,
+          lo_trans_file  TYPE REF TO zcl_abapgit_properties_file.
+
+
+    LOOP AT it_language INTO lv_langu.
+
+      ls_data = fill_translation( iv_name  = is_intf-vseointerf-clsname
+                                  iv_langu = lv_langu ).
+
+      " convert AFF type to JSON
+      TRY.
+          lo_ajson = zcl_abapgit_ajson=>new( iv_keep_item_order = abap_true
+            )->set( iv_path = '/'
+                    iv_val  = ls_data
+            )->map( zcl_abapgit_ajson_mapping=>create_to_camel_case( )
+            )->filter( zcl_abapgit_ajson_filter_lib=>create_empty_filter( ) ).
+          " remove manually the non-primitive types that are initial or not relevant for translation
+          lo_ajson->delete( '/category/' ).
+          lo_ajson->delete( '/proxy/' ).
```

Comment:
Just for my understanding: You remove fields "category" and "proxy" from the JSON because these are non-translatable fields?

Reply:
The filter
```
zcl_abapgit_ajson_filter_lib=>create_empty_filter( )
```
removes all empty fields. String empty is detected, however default enums, boolean and maybe others are not filtered. So this is done manually.

Reply:
Well, I prefer not to do it in this state.
When productising we could invest into an automatism. Considering the ENUM defaults, they are provided on ABAP side via annotation or in JSON Schema with the default field. So the information is either way there.